src/console_reporter.cc

```diff
@@ -152,21 +175,19 @@ void ConsoleReporter::PrintRunData(const Run& result) {
     printer(Out, COLOR_CYAN, "%10lld", result.iterations);
   }
 
-  for (auto& c : result.counters) {
-    const std::size_t cNameLen = std::max(std::string::size_type(10),
-                                          c.first.length());
-    auto const& s = HumanReadableNumber(c.second.value, c.second.oneK);
-    if (output_options_ & OO_Tabular) {
-      if (c.second.flags & Counter::kIsRate) {
-        printer(Out, COLOR_DEFAULT, " %*s/s", cNameLen - 2, s.c_str());
-      } else {
-        printer(Out, COLOR_DEFAULT, " %*s", cNameLen, s.c_str());
-      }
-    } else {
-      const char* unit = (c.second.flags & Counter::kIsRate) ? "/s" : "";
-      printer(Out, COLOR_DEFAULT, " %s=%s%s", c.first.c_str(), s.c_str(),
-              unit);
-    }
+  const bool tabular = (output_options_ & OO_Tabular);
+  const char* fmt = tabular ? " %*s" : " %3$s=%2$*1$s";
+  for (auto const& c : result.counters) { 
+    std::string value = (c.second.format == Counter::kSIUnits) ?
```

Comment:
I don't like how we are doing memory allocations for every counter printed

Reply:
I previously did a local version using a char[] preallocated on the stack, but wasn't quite sure if doing the same thing with strcat and strcpy onto the buffer was as readable as the append calls.
Would you prefer the preallocated char buffer?

Reply:
I would prefer the ugliest approach - just a lot of `printer` calls.
You will have to do that to make the format string a non-pointer anyway.

Reply:
Got it. I expect to end up with only 2 calls to `printer` since 2 are the different format strings. 
Does it make sense for you?

Reply:
Sounds plausible i suppose.